services/requester/cloud_kms_rotation_signer.go

```diff
@@ -0,0 +1,104 @@
+package requester
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/onflow/flow-go-sdk/crypto"
+	"github.com/onflow/flow-go-sdk/crypto/cloudkms"
+	"github.com/rs/zerolog"
+)
+
+var _ crypto.Signer = &CloudKMSKeyRotationSigner{}
+
+// CloudKMSKeyRotationSigner is a crypto signer that contains a pool of
+// `crypto.Signer`[1] objects, each of which is tied to a Cloud KMS
+// asymmetric signing key.
+// It keeps track of the signer/key combination that should be used for
+// the next incoming signing request. This allows for faster submission
+// of transactions to the network, due to a sequence number not being
+// reused between different keys used.
+// It also contains logic to queue up signature requests and in case
+// there are more transactions pending than the keys in the pool it
+// will wait for transactions to get executed so the new sequence key
+// can be obtained.
+// The signer is concurrency-safe.
+//
+// [1](https://github.com/onflow/flow-go-sdk/blob/master/crypto/cloudkms/signer.go#L37)
+type CloudKMSKeyRotationSigner struct {
+	mux        sync.RWMutex
+	kmsSigners []*cloudkms.Signer
+	index      int
+	signersLen int
+	logger     zerolog.Logger
+}
+
+// NewSignerForKeys returns a new CloudKMSKeyRotationSigner, for the
+// given slice of Cloud KMS keys.
+func NewSignerForKeys(
+	ctx context.Context,
+	client *cloudkms.Client,
+	keys []cloudkms.Key,
+	logger zerolog.Logger,
+) (*CloudKMSKeyRotationSigner, error) {
+	if len(keys) == 0 {
+		return nil, fmt.Errorf("no asymmetric signing keys provided")
+	}
+
+	kmsSigners := make([]*cloudkms.Signer, len(keys))
+	for i, key := range keys {
+		kmsSigner, err := client.SignerForKey(ctx, key)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"could not create signer for key with ID: %s: %w",
```

Comment:
```suggestion
				"could not create KMS signer for the key with ID: %s: %w",
```

Reply:
Updated in https://github.com/onflow/flow-evm-gateway/pull/282/commits/60e42d7d4c0058cb0231c92650073becb020c7a9